src/main/java/de/tum/in/www1/artemis/service/export/QuizExerciseWithSubmissionsExportService.java

```diff
@@ -0,0 +1,83 @@
+import de.tum.in.www1.artemis.domain.quiz.QuizExercise;
+import de.tum.in.www1.artemis.repository.QuizExerciseRepository;
+import de.tum.in.www1.artemis.service.FilePathService;
+import de.tum.in.www1.artemis.service.FileService;
+import de.tum.in.www1.artemis.service.archival.ArchivalReportEntry;
+
+@Service
+public class QuizExerciseWithSubmissionsExportService {
+
+    private final QuizExerciseRepository quizExerciseRepository;
+
+    private final ObjectMapper objectMapper;
+
+    private final DataExportQuizExerciseCreationService dataExportQuizExerciseCreationService;
+
+    private final FileService fileService;
+
+    private final FilePathService filePathService;
+
+    public QuizExerciseWithSubmissionsExportService(QuizExerciseRepository quizExerciseRepository, MappingJackson2HttpMessageConverter springMvcJacksonConverter,
+            DataExportQuizExerciseCreationService dataExportQuizExerciseCreationService, FileService fileService, FilePathService filePathService) {
+        this.quizExerciseRepository = quizExerciseRepository;
+        this.objectMapper = springMvcJacksonConverter.getObjectMapper();
+        this.dataExportQuizExerciseCreationService = dataExportQuizExerciseCreationService;
+        this.fileService = fileService;
+        this.filePathService = filePathService;
+    }
+
+    public Path exportExerciseWithSubmissions(QuizExercise exercise, Path exerciseExportDir, List<String> exportErrors, List<ArchivalReportEntry> reportEntries) {
+
+        var quizExercise = quizExerciseRepository.findByIdWithQuestionsAndStatisticsAndCompetenciesElseThrow(exercise.getId());
+        try {
+            fileService.writeObjectToJsonFile(exercise, objectMapper, exerciseExportDir.resolve("Exercise-Details-" + exercise.getSanitizedExerciseTitle() + ".json"));
+        }
+        catch (IOException e) {
+            exportErrors.add("Failed to export quiz exercise details " + exercise.getTitle() + " with id " + exercise.getId() + " due to a JSON processing error.");
+        }
+        List<Path> imagesToExport = new ArrayList<>();
+        for (var quizQuestion : quizExercise.getQuizQuestions()) {
+            if (quizQuestion instanceof DragAndDropQuestion dragAndDropQuestion) {
+                if (dragAndDropQuestion.getBackgroundFilePath() != null) {
+                    imagesToExport.add(filePathService.actualPathForPublicPath(URI.create(dragAndDropQuestion.getBackgroundFilePath())));
+                }
+                for (var dragItem : dragAndDropQuestion.getDragItems()) {
+                    if (dragItem.getPictureFilePath() != null) {
+                        imagesToExport.add(filePathService.actualPathForPublicPath(URI.create(dragItem.getPictureFilePath())));
+
+                    }
+                }
+                if (!imagesToExport.isEmpty()) {
+                    var imagesDir = exerciseExportDir.resolve("images-for-drag-and-drop-question-" + dragAndDropQuestion.getId());
```

Comment:
Can we save the background image separately or give it a special file name, so it can be differentiated from the other pictures?

Reply:
Should we also export the mapping of the elements to their corresponding gaps?

Reply:
The background image always has a file name "DragAndDropBackgroundImage-Something", so it can already be differentiated.
the mapping is already exported in the exercise-details.json